endoreg_db/services/lookup_service.py

```
@@ -89,25 +108,180 @@ def build_initial_lookup(pe: PatientExamination) -> Dict[str, Any]:
+        existing_lock = store.get('_recomputing_lock')
+        current_time = datetime.now()
+        
+        if existing_lock:
+            try:
+                lock_time = datetime.fromisoformat(existing_lock)
+                # Check if lock has expired
+                if current_time - lock_time < timedelta(seconds=LOCK_TTL_SECONDS):
+                    logger.warning(f"Recompute already in progress for token {token}, skipping")
+                    return {}
+                else:
+                    logger.info(f"Expired lock found for token {token}, proceeding with recompute")
```

Pylint (3.3.7)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)